### ISSUE TITLE
MAINT: decide minimum numpy dependency

### ISSUE THREAD

**Issue by bsipocz** [NASA-NAVO/navo-workshop#87]
Latest numpy and oldest astropy is not compatible (the reason for #86), but rather than bumping the minimum required astropy version, I would suggest to define a  minimum numpy instead, and it to be 1.18, which is the minimum numpy corresponding to the minimum astropy we have.

We can then always require a newer one if/when we run into compatibility issues.

**Comment by trjaffe** [comment]
I'm confused.  If the *latest* numpy is incompatible with the specified astropy, why would setting a minimum version help?  I'm missing something.  (And I'd like to understand it so I can fix my environment.)

**Comment by bsipocz** [comment]
We can support multiple user setups, e.g. one stating on LTS astropy is likely will have older numpy, too. And if nothing in the notebooks are using features  or fixes from a newer astropy/numpy then we can say we support these users. But, my very strong opinion, if we say we support them, then we should test for these scenarios. 

OTOH, these old versions may very well be not compatible with the latest and greatest, e.g. astropy 5.0 is incompatible with numpy 1.23. But that is fine, as I said above it's not expected that anyone would go with mixed versions like that (IMO I would like to see e.g. timestamped dependency checks, e.g. latest dependency version combination at a given date, but this approach is not exactly wide spread).

So, back to this issue: we have stated that we need astropy >=5.0, but didn't say anything for numpy. And that's wrong, as most often there are no upper limits set in configuration files for the libraries (a few tried that and it led to a huge mess), so we better also set a minimum numpy version, too and my suggestion of it to correspond to the minimum astropy version if possible (side note, I tried and it's not exactly possible if we also stay with requiring python 3.9, as there are no wheels available for that old numpy, and then we end up in a build nightmare situation, so I'll come up with a version that will also makes sense for CI and install convenience both for numpy and scipy).
